modin/backends/pandas/query_compiler.py

```diff
@@ -1862,15 +1862,68 @@ def mode_builder(df):
     def fillna(self, **kwargs):
         """Replaces NaN values with the method provided.
 
+        Parameters
+        ----------
+            squeeze_self : boolean
+                The flag to squeeze self.
+            squeeze_value : boolean
+                The flag to squeeze value (this flag is applied if value is query compiler and denotes a Series type).
+
         Returns:
             A new QueryCompiler with null values filled.
         """
+        squeeze_self = kwargs.pop("squeeze_self", False)
+        squeeze_value = kwargs.pop("squeeze_value", False)
         axis = kwargs.get("axis", 0)
         value = kwargs.get("value")
         method = kwargs.get("method", None)
         limit = kwargs.get("limit", None)
         full_axis = method is not None or limit is not None
-        if isinstance(value, dict):
+        if isinstance(value, BaseQueryCompiler):
+            kwargs.pop("value")
+            if squeeze_self:
+                if full_axis:
+                    value = value.to_pandas().squeeze(axis=1)
+
+                    def fillna_builder(series):
+                        # Limit parameter works only on Series type, so we have to squeeze both objects to get
```

Comment:
```suggestion
                        # `limit` parameter works only on `Series` type, so we have to squeeze both objects to get
```

Reply:
Ok. Although I am not sure quotes in non-documentation comments are seen anywhere.

Reply:
That looks more highlighted anyway.